File: subtract-the-product-and-sum-of-digits-of-an-integer.rb
# @param {Integer} n
# @return {Integer}
def subtract_product_and_sum(n)
  numbers = n.to_s.split('').map{ |num| num.to_i }
  p numbers.inject(:*) - numbers.inject(:+)
end

subtract_product_and_sum(234)

File: js/subtract-the-product-and-sum-of-digits-of-an-integer.js
/**
 * @param {number} n
 * @return {number}
 */
var subtractProductAndSum = function(n) {
  let numbers = n.toString().split('').map(Number);
  let sum = numbers.reduce((a, b) => a + b);
  let product = numbers.reduce((a, b) => a * b, 1);
  return product - sum;
};

console.log(subtractProductAndSum(234));
